nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/SendTrapSNMP.java

```diff
@@ -0,0 +1,187 @@
+            .name("snmp-trap-manager-address")
+            .displayName("Manager (destination) IP address")
+            .description("The address of the SNMP manager where the trap is being sent to.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor SYSTEM_UPTIME = new PropertyDescriptor.Builder()
+            .name("snmp-trap-sysUpTime")
+            .displayName("System uptime")
+            .description("The system uptime.")
+            .required(true)
+            .addValidator(StandardValidators.INTEGER_VALIDATOR)
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("All FlowFiles that are received from the SNMP agent are routed to this relationship")
+            .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("All FlowFiles that cannot received from the SNMP agent are routed to this relationship")
+            .build();
+
+    protected static final List<PropertyDescriptor> PROPERTY_DESCRIPTORS = Collections.unmodifiableList(Arrays.asList(
+            SNMP_CLIENT_PORT,
+            AGENT_HOST,
+            AGENT_PORT,
+            SNMP_VERSION,
+            SNMP_COMMUNITY,
+            SNMP_SECURITY_LEVEL,
+            SNMP_SECURITY_NAME,
+            SNMP_AUTH_PROTOCOL,
+            SNMP_AUTH_PASSWORD,
+            SNMP_PRIVACY_PROTOCOL,
+            SNMP_PRIVACY_PASSWORD,
+            SNMP_RETRIES,
+            SNMP_TIMEOUT,
+            ENTERPRISE_OID,
+            AGENT_ADDRESS,
+            GENERIC_TRAP_TYPE,
+            SPECIFIC_TRAP_TYPE,
+            TRAP_OID,
+            MANAGER_ADDRESS,
+            TRAP_OID_VALUE,
+            SYSTEM_UPTIME
+    ));
+
+    private static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+            REL_SUCCESS,
+            REL_FAILURE
+    )));
+
+    private SNMPTrapSender snmpTrapSender;
+
+    @OnScheduled
+    @Override
+    public void initSnmpClient(ProcessContext context) {
+        super.initSnmpClient(context);
+        snmpTrapSender = new SNMPTrapSender(snmpClient, target);
+    }
+
+    /**
+     * Delegate method to supplement
+     * {@link #onTrigger(ProcessContext, ProcessSession)}. It is implemented by
+     * sub-classes to perform {@link Processor} specific functionality.
+     *
+     * @param context        instance of {@link ProcessContext}
+     * @param processSession instance of {@link ProcessSession}
+     * @throws ProcessException Process exception
+     */
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession processSession) {
+        final String enterpriseOIDValue = context.getProperty(ENTERPRISE_OID).getValue();
```

Comment:
Would not make sense to extract these properties during onSchedule and depend on those later?